base/point.h

```diff
@@ -25,13 +25,19 @@ class Point {
    * \brief Standard destructor.
    */
   virtual ~Point();
-  /** General accessor method for the underlying data.
+
+  /** General accessor method for each feature of the underlying data.
    * \param feature_id The coordinate of interest. Indexing is implementation
    * specific.
    * \return The value of the coordinate specified in the input.
    */
   virtual double GetFeatureValue(unsigned feature_id) const = 0;
 
+  /** General accessor method for the label of the underlying data.
+   * \return The label of the object.
+   */
+  virtual double GetLabel() const = 0;
```

Comment:
What I was thinking of the interface was to create the most generic structure possible, so any functions that are more specific to some kind of problem would be implemented by classes specialized in this problems. 
But it is more of a question then a problem per se, as I'm not used to work with big implementations and projects like this.